docs/custom-plugins/index.md

```diff
@@ -0,0 +1,81 @@
+
+## Extend Redocly with custom plugins
+
+Custom plugins can extend the built-in functionality of Redocly in the following ways:
+
+- **Write your own custom rules in JavaScript**. If your API standards include elements that can't be expressed by our existing rules, including configurable ones, you can code your own rules. See the [documentation on writing your own rules in custom plugins](./custom-rules.md) for more information.
+
+- **Define configuration in a custom plugin**. Redocly supports [resuable ruleset configuration](../guides/configure-rules.md#create-reusable-configuration) already, but defining configuration in a custom plugin is particularly useful when the configuration belongs alongside other plugin elements. See the [documentation on configuration in plugins](./custom-config.md) for more information.
+
+- **Extend existing standard definitions with additional type definitions**. Working with extensions to defined standards can be helped by [extending the supported types](./extended-types.md).
+
+- **Write custom decorators in JavaScript**. Not all API descriptions are exactly as we'd like them to be before passing them to the next stage of the API lifecycle. Redocly has a selection of [decorators available](../decorators), but if you need to build something more then [visit the documentation for building decorators in custom plugins](./custom-decorators.md).
+
+
+### Order of execution
+
+Some commands, such as using `bundle` with `--lint`, may execute more than one type of functionality. The order is preprocessors first, then linting rules, and finally the decorators are applied.
+
+```mermaid
+graph LR
+    A(1. preprocessors) ==> B(2. rules)
+    B ==> V(3. decorators)
+    style A fill:#codaf9,stroke:#0044d4,stroke-width:4px
+    style B fill:#codaf9,stroke:#0044d4,stroke-width:4px
+    style V fill:#codaf9,stroke:#0044d4,stroke-width:4px
+```
+
+
+## Add custom plugins to your project
+
+Each plugin can contain any or many of each type of extension (rules, configuration, decorators, etc). 
+
+Define the plugins to include in the `redocly.yaml` configuration file, in the `plugins` section:
+
+```yaml
+
+plugins:
+  - plugins/my-best-plugin.js
+  - plugins/another-plugin.js
+```
+
+The paths are relative to the configuration file location. Where there are multiple features that happen at the same time, such as decorators, the plugins loaded first in the `plugins` section are processed first.
+
+
+### Plugin structure
+
+The minimal plugin should export an `id` string that is used to refer to the contents of the plugin in the `redocly.yaml` configuration file:
+
+```js
+module.exports = {
+  id: 'my-local-plugin'
+}
+```
+
+<a id="oas-major-versions"></a>
```

Comment:
```suggestion
```
Suggestion: remove anchor because there are no links to this document or anchor that I could find. Maybe I'm  searching the wrong place?